src/v/raft/replicate_batcher.h

```diff
@@ -27,12 +29,10 @@ class replicate_batcher {
         ss::promise<result<replicate_result>> _promise;
         replicate_result ret;
         size_t record_count;
+        ss::semaphore_units<> units;
```

Comment:
this deserves a comment on lifetime of the item

Reply:
done